vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -284,49 +323,53 @@ func (d *VolumeDriver) Create(r volume.Request) volume.Response {
 		// in which case we continue creating the file system.
 		fs.DevAttachWait(watcher, r.Name, device)
 	}
-	errMkfs := fs.Mkfs(mkfscmd, r.Name, device)
-	if errMkfs != nil {
-		log.WithFields(log.Fields{"name": r.Name,
-			"error": errMkfs}).Error("Create filesystem failed, removing the volume ")
-		errDetach := d.ops.Detach(r.Name, nil)
-		if errDetach != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errDetach}).Warning("Detach volume failed ")
+	if err := fs.Mkfs(mkfscmd, r.Name, device); err != nil {
+		log.WithField("name", r.Name).WithError(err).Error(
+			"Create filesystem failed, removing the volume")
+		if err := d.ops.Detach(r.Name, nil); err != nil {
+			log.WithField("name", r.Name).WithError(err).Warn(
+				"Detach volume failed")
 		}
-		errRemove := d.ops.Remove(r.Name, nil)
-		if errRemove != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errRemove}).Warning("Remove volume failed ")
+		if err := d.ops.Remove(r.Name, nil); err != nil {
+			log.WithField("name", r.Name).WithError(err).Warn(
+				"Remove volume failed")
 		}
-		return volume.Response{Err: errMkfs.Error()}
+		return volume.Response{Err: err.Error()}
```

Comment:
Pls. retain the individual errors - errDetach, errRemove. The idea is to return the core failure - mkfs error to the user. The errors that happen when handling the mkfs error to remove the volume are good for logging but isn't the core issue that needs to be returned.